File: classifier/internal/mlclientv2/schemas.go
// Package mlclientv2 provides a unified Go client for ML sidecar services
// that speak the standard response envelope.
package mlclientv2

import "encoding/json"

// StandardResponse is the envelope returned by POST /v1/classify on every ML sidecar.
type StandardResponse struct {
	Module           string          `json:"module"`
	Version          string          `json:"version"`
	SchemaVersion    string          `json:"schema_version"`
	Result           json.RawMessage `json:"result"`
	Relevance        *float64        `json:"relevance"`
	Confidence       *float64        `json:"confidence"`
	ProcessingTimeMs float64         `json:"processing_time_ms"`
	RequestID        string          `json:"request_id"`
}

// StandardError is the envelope returned on error responses from ML sidecars.
type StandardError struct {
	Error     string `json:"error"`
	Message   string `json:"message"`
	Module    string `json:"module"`
	RequestID string `json:"request_id"`
	Timestamp string `json:"timestamp"`
}

// HealthResponse is the envelope returned by GET /v1/health on every ML sidecar.
type HealthResponse struct {
	Status        string          `json:"status"`
	Module        string          `json:"module"`
	Version       string          `json:"version"`
	SchemaVersion string          `json:"schema_version"`
	ModelsLoaded  bool            `json:"models_loaded"`
	UptimeSeconds float64         `json:"uptime_seconds"`
	Checks        map[string]bool `json:"checks,omitempty"`
}

// classifyRequest is the POST body sent to /v1/classify.
type classifyRequest struct {
	Title    string         `json:"title"`
	Body     string         `json:"body"`
	Metadata map[string]any `json:"metadata,omitempty"`
}

File: ml-framework/nc_ml/schemas.py
"""Standard schemas for NC ML sidecar modules."""

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict


class ClassifyRequest(BaseModel):
    """Standard request for classification or extraction."""

    title: str
    body: str
    metadata: dict[str, Any] | None = None


class ModuleResult(BaseModel):
    """Base result model. Subclasses must not add extra fields."""

    model_config = ConfigDict(extra="forbid")


class ClassifierResult(ModuleResult):
    """Result from a classifier module."""

    relevance: float
    confidence: float


class ExtractorResult(ModuleResult):
    """Result from an extractor module."""


class StandardResponse(BaseModel):
    """Envelope returned by every ML sidecar endpoint."""

    module: str
    version: str
    schema_version: str
    result: ModuleResult
    relevance: float | None
    confidence: float | None
    processing_time_ms: float
    request_id: str


class StandardError(BaseModel):
    """Error envelope for failed requests."""

    error: str
    message: str
    module: str
    request_id: str
    timestamp: datetime


class HealthResponse(BaseModel):
    """Response from the health endpoint."""

    status: Literal["healthy", "degraded", "unhealthy"]
    module: str
    version: str
    schema_version: str
    models_loaded: bool
    uptime_seconds: float
    checks: dict[str, bool] | None = None
